src/main/app/access-code/AccessCodePostController.ts

```diff
@@ -25,29 +25,37 @@ export class AccessCodePostController {
 
     const { saveAndSignOut, saveBeforeSessionTimeout, _csrf, ...formData } = form.getParsedBody(req.body);
 
-    formData.respondentUserId = req.session.user.id;
-    formData.applicant2Email = req.session.user.email;
+    //ToDo: confirm setting the first and last names from session is
+    //how it should work, we were doing this on joint application previously
+    //but oddly not on respondent
+    if (req.path.includes(APPLICANT_2)) {
+      formData.applicant2Email = req.session.user.email;
+      formData.applicant2FirstNames = req.session.user.givenName;
+      formData.applicant2LastNames = req.session.user.familyName;
+      formData.respondentUserId = req.session.user.id;
+    } else {
+      formData.applicant1Email = req.session.user.email;
+      formData.applicant1FirstNames = req.session.user.givenName;
+      formData.applicant1LastNames = req.session.user.familyName;
+      formData.applicant1UserId = req.session.user.id;
```

Comment:
Won't this set applicant 1s name and details when the respondent logs in and the path includes "respondent" rather than "APPLICANT_2"?

Reply:
Thanks Adam. The Oidc middleware when detects presence of an invite code for the logged in user and no existing linked case redirects to appropriate url to key in access code. The respondent flow uses the same url as applicant2.
if (req.session.user.email === newInviteUserCase.applicant1Email) {
            redirectUrl = `${APPLICANT_1}${ENTER_YOUR_ACCESS_CODE}`;
          } else {
            redirectUrl = `${APPLICANT_2}${ENTER_YOUR_ACCESS_CODE}`;
          }

I have tested this with respondent journey and it works fine. But can leave a note on ticket for QA to test this too.

Reply:
Sure. Will update the ticket for Elis to test this.
My own observation has been this.
When I put the respondent url --> http://localhost:3001/respondent in browser, it first directs me to login. Once logged in, the system redirects to entering access code (and url is /applicant2/enter-your-access-code). If then however, the user changes the url to "/respondent" again, the system does nothing and stays on the same page but any access code and case reference entered will fail validation.